### DISCUSSION TITLE
sample image quality

### DISCUSSION THREAD

**Discussion by zhuliyi0** [bghira/SimpleTuner#110] [General]
Well this is more of a question than issue report. In all my experiment runs with different datasets, the sample image always has a feel of "blotchy-ness", right from the start to the end. It gets a little better at late epochs, but never goes away. To rule out potential causes from my training parameters, I run a train with lr 1e-9, sample at 1st step and save model. The sample still has that blotchy feel. I then used that model in comfy to generate some images with the same prompt, and the inference images are absolutely not blotchy. Other than that sample images are very similar to inference images.

I forgot to save these examples , will generate them again if needed for this discussion. Also the datasets I have tested are of same subject category, maybe need to try more training subject types. 

Wonder what could be the cause.

**Comment by bghira** [comment]
in Weights and Biases? I notice the images look remarkably bad there.

**Comment by bghira** [comment]
here is one of my validation samples.

![image](https://github.com/bghira/SimpleTuner/assets/59658056/4c2cdf82-32c8-4996-9c85-ad91394e1423)

**Comment by zhuliyi0** [comment]
> in Weights and Biases? I notice the images look remarkably bad there.

Could you be more specific? I am not getting it.

**Comment by bghira** [comment]
wandb.ai tracking site

**Comment by zhuliyi0** [comment]
> wandb.ai tracking site

Oh that, no I open image directly in jupyter. Them look the same as on wandb.

**Comment by zhuliyi0** [comment]
your sample image looks great. I suspect it has something to do with my training subject, which is something SDXL base model is terrible at, but this does not explain why comfy does not look blotchy, although the content is same as bad. Will do more experiment.

**Comment by bghira** [comment]
i've done thousands of steps on v-prediction and terminal SNR with this image, to specifically address blotchy images which to me seem to be caused by offset noise in the base model

**Comment by bghira** [comment]
blotchy SDXL example:

![image](https://github.com/bghira/SimpleTuner/assets/59658056/9236dbc6-279b-4c38-a6cb-0231ea409788)

**Comment by zhuliyi0** [comment]
It turns out to be the scheduler difference. The default ddim scheduler outputs bad quality while euler-a is much better. I tried changing the scheduler but didn't do it right the first time. Now the samples are very close to inference's quality. Case closed. Thanks for help!

**Comment by bghira** [comment]
Euler A is pretty unstable, and it's not very indicative of whether the model works or not, but if it's doing what you need then arguably it's fine for your local use case.

DDIM is likely not doing as well as Euler A because of the default DDIM config.

try `ptx0/sdxl-base` instead of `stabilityai/stable-diffusion..`

**Comment by zhuliyi0** [comment]
That's good info, I did use SAI's huggingface version of SDXL, and now I see it comes with a scheduler config for Euler, but not DDIM, and there is one setting that is different. Will give it a try. That might explain why Euler A worked much better.

**Comment by zhuliyi0** [comment]
No I tried and it did not make a difference.

I can see DDIM's config.json uses "trailing" setting and Euler's uses "leading", other than that there is no difference between the two scheduler_config.json files. I didn't change the default arg setting of "trailing" either, so Euler-A was using "trailing" and that didn't cause noticable issue. Maybe there are other things that I am missing.. 

I didn't download the full model from ptx0/sdxl-base, only replaced that json file. Or do I need to replace more file/do a full download?

**Comment by bghira** [comment]
beyond that, my `ptx0/sdxl-base` model also has a scheduler config that enables rescaled betas for zero terminal SNR.
